src/pymorize/cmorizer.py

```diff
@@ -288,6 +294,25 @@ def _check_is_subperiod(self):
         if errors:
             for err in errors:
                 logger.error(err)
+            raise errors[0]
+
+    def _check_units(self):
+        errors = []
+        for rule in self.rules:
+            for input_collection in rule.inputs:
+                try:
+                    filename = input_collection.files[0]
+                except IndexError:
+                    break
+                da = xr.open_dataarray(filename, use_cftime=True)
+                try:
+                    handle_unit_conversion(da, rule)
```

Comment:
okay. will rewrite it.

Reply:
Question: how to know the unit in the netcdf file without accessing it. I can check the `cmor_variable` (table unit) alone from the rule object if it dimensionless and if the unit mapping is found in dimensionless_units but how about the source units. If `model_variable` is defined then I can check it as well but that means the user wants to use `model_variable` instead of the units found in the netcdf file.

consider the following table

| var.name     | modelunit  | tableunit | converted_representation |
|--------------|------------|-----------|--------------------------|
| sidmasstranx | kg s-1 m-1 | kg s-1    |                          |
| sisnconc     | 1          | %         | 100.0 percent            |
| sitimefrac   | 1.0        | 1         | 1.0 dimensionless        |
| so           | psu        | 0.001     | 1 dimensionless          |

If I just check the `tableunit`, then I can not catch `sidmasstranx` failing and for `sitimefrac` the check will fail as it is dimensionless and there is no entry in the mapping for it yet if the user puts an mapping entry for this variable then it still fails as source units (from netcdf file or user defined modelunits) are still dimensionless and the unit conversion function does not consider using mapping for source units.

Reply:
The re-written code only deals with dimensionless variables (i.e., ignoring `sidmasstranx` case) and addresses the rest of the cases.

Reply:
> Question: how to know the unit in the netcdf file without accessing it. I can check the cmor_variable (table unit) alone from the rule object if it dimensionless and if the unit mapping is found in dimensionless_units but how about the source units. If model_variable is defined then I can check it as well but that means the user wants to use model_variable instead of the units found in the netcdf file.

I think this is a question for when we prepare the validation schema #71

> then I can not catch sidmasstranx failing

This is for another PR, to me it looks like an integration over distance. That's also for another issue and feature, where I presume we will need to load some dimension of the element to be able to integrate over the distance? Are the fluxes over the element interfaces linear? This could get messy really fast...

> sitimefrac the check will fail as it is dimensionless and there is no entry in the mapping for it yet

That's okay, there is an issue to complete the mappings: #73 

> but if the user puts an mapping entry for this variable then it still fails as source units (from netcdf file or user defined modelunits) are still dimensionless and the unit conversion function does not consider using mapping for source units.

That is intended, the user needs in that case to specify the units in their yaml configuration file, and now they get a nice error message telling them exactly what they need to do :)